### PR TITLE
Dfl 9395 9396 profile cards

### PR DESCRIPTION
**Pull request by kevinhowbrook**
# Description

This is based off https://github.com/mozilla/foundation.mozilla.org/pull/9437

This PR adds a new Profile block to the main streamfield body field:

Screenshot(21)

The backend looks like this:

https://user-images.githubusercontent.com/2128707/193584563-09b4141c-ead2-48bb-84ef-088d498eb076.mp4

Link to sample test page: http://localhost:8000/en/data-future-labs/
Related PRs/issues: #9395 and #9396 

# Checklist

**Tests**
- [ ] ~~Is the code I'm adding covered by tests?~~ Tests for this are already covered by wagtail

**Changes in Models:**
- [ ] ~~Did I update or add new fake data?~~
- [ ]  ~~Did I squash my migration?~~
- [ ]  ~~Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~ New feature so as I understand it, this is not needed

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

**Comment by tbrlpld** [comment]
@kevinhowbrook I don't think this is documented anywhere... but I think it would be great to add new streamfield blocks to the styleguide factory (`network-api/networkapi/wagtailpages/factory/styleguide.py`) and add a provider function (`network-api/networkapi/utility/faker/streamfield_provider.py`). This should add the new blocks to the style guide (http://localhost:8000/en/style-guide/) that is generated when new fake content is loaded `inv new-db`.

The style guide is also covered by visual regression tests, which should make it possible for us to catch accidental regression on those new blocks.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{% extends "./base_streamfield_block.html" %}
+{% load wagtailcore_tags wagtailimages_tags %}
+
+{% block block_content %}
+	<div class="tw-row tw-py-4">
+		{% for block in self.profiles %}
+          {% with card=block.profile count=self.cards|length %}
+            {% image card.image fill-140x140 as img %}
+            <div class="tw-px-4 small:tw-w-full {% if count > 2 %} large:tw-w-1/3 medium:tw-w-1/3 {% else %} medium:tw-w-1/3 {% endif %} tw-mb-5">
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html:9]: **Question** Can you add some context why we need the if statement? It seems like we apply `medium:tw-w-1/3` in anycase. Also, the width we set for the "large and above"breakpoint sets the same width as the "medium and above" breakpoint and should not be necessary.

**Review by tbrlpld — COMMENTED** [review]
@kevinhowbrook I left a few questions around the template. 

I noticed it looks really tight on tablet when there is also an aside. What do you think about a `min-width` on the individual cards to make them wrap to the newline? 

image

**Comment by kevinhowbrook** [comment]
> @kevinhowbrook I left a few questions around the template.
> 
> I noticed it looks really tight on tablet when there is also an aside. What do you think about a `min-width` on the individual cards to make them wrap to the newline?
> 
> image

This should be resolved now it's been rebased - the layout responds better to accommodate the 3 profiles:


https://user-images.githubusercontent.com/2128707/194498616-496fd532-9fef-464c-9aef-2aff63852061.mp4

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/profile.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from wagtail.snippets import blocks as snippet_blocks
+from networkapi.wagtailpages.pagemodels.profiles import Profile
+
+
+class ProfileCard(blocks.StructBlock):
+    profile = snippet_blocks.SnippetChooserBlock(Profile)
+
+
+class ProfileBlock(blocks.StructBlock):
+    profiles = blocks.ListBlock(ProfileCard())
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/profile.py]: added in 9ab09b237b2e869543e617301269387e26ab57ce :+1:

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+              <div class="card-regular tw-h-full tw-flex-col">
+                {% if card.image %}
+                    <img src="{{ img.url }}" alt="{{ card.alt_text }}" class="tw-rounded-full tw-w-[140px] tw-h-[140px]" />
+                {% endif %}
+                <div class="tw-flex tw-flex-1">
+                  <div class="tw-bg-white tw-relative tw-py-4 tw-w-full tw-flex tw-flex-col">
+                    <h5 class="tw-h5-heading tw-mb-2">{{ card.name }}</h3>
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html:16]: :+1: swapped in e9bb10b32616dedc9c2862455fb15337f8aa5c0b

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{% extends "./base_streamfield_block.html" %}
+{% load wagtailcore_tags wagtailimages_tags %}
+
+{% block block_content %}
+	<div class="tw-row tw-py-4">
+		{% for block in self.profiles %}
+          {% with card=block.profile count=self.cards|length %}
+            {% image card.image fill-140x140 as img %}
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html:8]: It won't throw an error but I'll shift it inside the if check, makes way more sense to conditionally set the image, thanks

[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html]: eee0322f0042ac1574a4aa173943ae33fb0debab

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html (hunk 1 above)
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/profile_block_rounded.html]: Again, these aren't needed - I thought they were, so I've removed them, thanks! 1bda82b11d9db204639d8875b7fb32d2bdf9aa54

**Comment by kevinhowbrook** [comment]
> @kevinhowbrook I don't think this is documented anywhere... but I think it would be great to add new streamfield blocks to the styleguide factory (`network-api/networkapi/wagtailpages/factory/styleguide.py`) and add a provider function (`network-api/networkapi/utility/faker/streamfield_provider.py`). This should add the new blocks to the style guide (http://localhost:8000/en/style-guide/) that is generated when new fake content is loaded `inv new-db`.
> 
> The style guide is also covered by visual regression tests, which should make it possible for us to catch accidental regression on those new blocks.

This is a good idea and one we should do for all the new blocks, I think we should add a new issue for this as I'm too tight on time

**Comment by tbrlpld** [comment]
> This is a good idea and one we should do for all the new blocks, I think we should add a new issue for this as I'm too tight on time

Doing this in a new ticket sounds good to me, given the time constraints.

**Review by tbrlpld — APPROVED** [review]

**Pull request merged**